.github/workflows/verify-build.yml

```diff
@@ -0,0 +1,69 @@
+# This workflow will perform a test whenever there
+# is some change in code done to ensure that the changes
+# are not buggy and we are getting the desired output.
+name: Test Build
+on: [push, pull_request]
+jobs:
+  build:
+    name: Build image using Buildah
+    runs-on: ubuntu-latest
+    steps:
+      # Fetch name of the Forked Repository
+      - name: Fetch PR head repo
+        if: github.event_name == 'pull_request'
+        id: head_repo_name
+        run: |
+          HEAD_REPO_NAME=$(jq -r '.pull_request.head.repo.full_name' "$GITHUB_EVENT_PATH")
+          echo "PR head repo: $HEAD_REPO_NAME"
+          echo "::set-output name=repo::$HEAD_REPO_NAME"
+      # Extract branch name
+      - name: Extract branch name
+        if: github.event_name != 'pull_request'
+        shell: bash
+        run: echo "branch=${GITHUB_REF#refs/heads/}" >> $GITHUB_ENV
+      # Extract branch name on pull request to checkout
+      # the Buildah action from that branch
+      - name: Extract branch name on pull request
+        if: github.event_name == 'pull_request'
+        run: echo "branch=${GITHUB_HEAD_REF}" >> $GITHUB_ENV
+      - name: Checkout Buildah action
+        uses: actions/checkout@v2
+        with:
+          repository: ${{ steps.head_repo_name.outputs.repo }}
+          ref: ${{ env.branch }}
+          path: "buildah-action"
+      - name: Checkout spring-petclinic project
+        uses: actions/checkout@v2
+        with:
+          repository: "spring-projects/spring-petclinic"
+          path: "spring-petclinic"
+      # Setup java.
+      - name: Setup Java
+        uses: actions/setup-java@v1
+        with:
+          java-version: 11
+      # Run maven to build the project
+      - name: Maven
+        run: |
+          # navigate to spring-petclinic repository
+          cd spring-petclinic
+          mvn package
+      # Navigate back to home directory
+      - name: Change directory
+        run: cd ..
+      # Build image using Buildah action
+      - name: Build Image
+        uses: ./buildah-action/
+        with:
+          image: spring-petclinic
+          tag: v1
+          content: |
+            ./spring-petclinic/target/spring-petclinic-2.3.0.BUILD-SNAPSHOT.jar
+          entrypoint: |
+            java
+            -jar
+            spring-petclinic-2.3.0.BUILD-SNAPSHOT.jar
+          port: 8080
+      # Check if image is build
+      - name: Check images created
+        run: buildah images | grep 'spring-petclinic'
```

Comment:
This step should fail if the output is empty.

Reply:
eg.

```
run: |
  export NEW_IMAGE=$(buildah images | grep 'spring-petclinic')
  echo "New image: $NEW_IMAGE"
  [[ -n $NEW_IMAGE ]] # This will exit with 1 if $NEW_IMAGE is blank

Reply:
Actually above step also used to fail with exit code 1, I tried that earlier.

Because output is blank when there is no image of `spring-petclinic"

Reply:
What exactly would fail? grep?

Reply:
TIL grep exits 1 if there are no matches. cool